sklearn/neural_network/extreme_learning_machines.py

```diff
@@ -0,0 +1,629 @@
+
+        y : array-like, shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like, shape (n_samples,)
+            Per-sample weights. Rescale C per sample. Higher weights
+            force the classifier to put more emphasis on these points.
+
+        Returns
+        -------
+        self : returns a trained ELM usable for prediction.
+        """
+        return self._fit(X, y, sample_weight=sample_weight, incremental=False)
+
+    def partial_fit(self, X, y, sample_weight=None):
+        """Fit the model to the data X and target y.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Subset of training data.
+
+        y : array-like, shape (n_samples,)
+            Subset of target values.
+
+        sample_weight : array-like, shape (n_samples,)
+            Per-sample weights. Rescale C per sample. Higher weights
+            force the classifier to put more emphasis on these points.
+
+        Returns
+        -------
+        self : returns a trained ELM usable for prediction.
+        """
+        self._fit(X, y, sample_weight=sample_weight, incremental=True)
+
+        return self
+
+    def _decision_scores(self, X):
+        """Predict using the ELM model
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            The input data.
+
+        Returns
+        -------
+        y_pred : array-like, shape (n_samples,) or (n_samples, n_outputs)
+            The predicted values.
+        """
+        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
+
+        if self.batch_size is None:
+            hidden_activations = self._compute_hidden_activations(X)
+            y_pred = safe_sparse_dot(hidden_activations, self.coef_output_)
+        else:
+            n_samples = X.shape[0]
+            batches = gen_batches(n_samples, self.batch_size)
+
+            y_pred = np.zeros((n_samples, self.n_outputs_))
+            for batch in batches:
+                h_batch = self._compute_hidden_activations(X[batch])
+                y_pred[batch] = safe_sparse_dot(h_batch, self.coef_output_)
+
+        return y_pred
+
+
+class ELMClassifier(BaseELM, ClassifierMixin):
+    """Extreme learning machine classifier.
+
+    The algorithm trains a single-hidden layer feedforward network by computing
+    the hidden layer values using randomized parameters, then solving
+    for the output weights using least-square solutions. For prediction,
+    after computing the forward pass, the continuous output values pass
+    through a gate function converting them to integers that represent classes.
+
+    This implementation works with data represented as dense and sparse numpy
+    arrays of floating point values for the features.
+
+    Parameters
+    ----------
+    C : float, optional, default 100
+        A regularization term that controls the linearity of the decision
```

Comment:
To be more precise, in LS-SVM, they define a constrained form of the objective then derive a dual in order to introduce kernels. In ridge regression, we usually introduce kernels via the representer theorem but the resulting objective is the same.